lib/griffin/interceptors/server/sentry_interceptor.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+gem "sentry-ruby"
+
+module Griffin
+  module Interceptors
+    module Server
+      class SentryInterceptor < GRPC::ServerInterceptor
+        TRANSACTION_OP = "griffin"
+
+        def request_response(request: nil, call: nil, method: nil)
+          return yield unless Sentry.initialized?
+
+          Sentry.clone_hub_to_current_thread
+
+          Sentry.with_scope do |scope|
+            Sentry.with_session_tracking do
+              scope.clear_breadcrumbs
+
+              service_name = call.service_name
+              method_name = method.name
+              scope.set_transaction_name("#{service_name}/#{method_name}")
+
+              transaction = start_transaction(call.metadata, scope)
+              scope.set_span(transaction) if transaction
+
+              if call.metadata["x-request-id"]
+                scope.set_tags(request_id: call.metadata["x-request-id"])
+              end
+
+              begin
+                yield
+              rescue => e
+                if e.is_a?(GRPC::BadStatus)
+                  finish_transaction(transaction, e.code)
+                  raise e
+                end
+
+                GRPC.logger.error("Internal server error: #{e}")
+                finish_transaction(transaction, GrpcKit::StatusCodes::INTERNAL)
```

Comment:
Even if it raises in a rescue block, codes in an ensure block always runs, so does `finish_transaction` call doubly, right?

```ruby main.rb
begin
  raise 'a'
rescue
  raise 'b'
ensure
  p 'ensure'
end
```

```console
hfm@local ~> ruby main.rb
"ensure"
main.rb:4:in `rescue in <main>': b (RuntimeError)
        from main.rb:3:in `<main>'
main.rb:2:in `<main>': a (RuntimeError)
```

Reply:
Oops, I didn't give it enough consideration. fixed: https://github.com/cookpad/griffin-interceptors/pull/14/commits/d23766b69080ed26a9ef537dce897e8803f98746